opencog/attention/WARentCollectionAgent.cc

```diff
@@ -56,7 +57,7 @@ Handle WARentCollectionAgent::tournamentSelect(HandleSeq population){
         tournament[i] = population[idx];
     }
 
-    auto result = std::max_element(tournament, tournament + (_tournamentSize - 1), []
+    auto result = std::max_element(tournament, tournament + (sz - 1), []
```

Comment:
But there is nothing that checks this in this function. At least throw an exception so that we know when that happens.